### PR TITLE
[DOCS] Add Getting started guide

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
After Berlin docs restructure, this re-adds new getting started guide. Next steps in progress.

### CODE REVIEW

File: docs/_getting-started/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+_[Read this guide](/interface/Managing-your-accounts) for more details on importing existing Ethereum accounts and other uses of the `account` command._
+
+### Sync modes
+
+Running Geth starts an Ethereum node that can join any existing network, or create a new one. You can start Geth in one of three different sync modes using the `--syncmode "{mode}"` argument that determines what sort of node it is in the network.
+
+These are:
+
+-   **Full**: Downloads all block headers, block data, and validates all transactions
+-   **Fast** (Default): Downloads block headers and block data of the most recent transactions (1024) and validates them.
```
Comment (holiman) [inline docs/_getting-started/index.md]: Nope. Fast downloads all blocks ( headers + transactions + receipts), verifies all headers, anda also downloads the state and verifies against the headers.

File: docs/_getting-started/index.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+_[Read this guide](/interface/Managing-your-accounts) for more details on importing existing Ethereum accounts and other uses of the `account` command._
+
+### Sync modes
+
+Running Geth starts an Ethereum node that can join any existing network, or create a new one. You can start Geth in one of three different sync modes using the `--syncmode "{mode}"` argument that determines what sort of node it is in the network.
+
+These are:
+
+-   **Full**: Downloads all block headers, block data, and validates all transactions
```
Comment (holiman) [inline docs/_getting-started/index.md]: Full download all blocks ( headers + transactions + receipts), and generates the 'state' incrementally by executing every block one by one

File: docs/_getting-started/index.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+-   **Fast** (Default): Downloads block headers and block data of the most recent transactions (1024) and validates them.
+-   **Light**: Downloads all block headers, block data, but does not validate transactions.
```
Comment (holiman) [inline docs/_getting-started/index.md]: Not quite. Downloads all block headers, and verify some of them randomly. Fetches portions of state on demand as needed.

File: docs/_getting-started/index.md (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/_getting-started/index.md]: Thanks @holiman here's my change based on your comments. I think I'd like to add a clarification here of what you mean exactly by "state". Bearing in mind this is a getting started guide, what does it mean to you?

```suggestion
-   **Fast** (Default): Downloads all blocks (including headers, transactions and receipts), verifies all headers, and downloads the state and verifies it against the headers.
```

Comment (ChrisChinchilla) [reply docs/_getting-started/index.md]: Or maybe the state question is answered below…?

File: docs/_getting-started/index.md (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/_getting-started/index.md]:
```suggestion
-   **Full**: Downloads all blocks (including headers, transactions and receipts) and generates the state of the blockchain incrementally by executing every block.
```

And again @holiman thanks, but can I unpack what "executing" mean?

I also think we might need a glossary of headers, transactions and receipts mean exactly.

File: docs/_getting-started/index.md (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/_getting-started/index.md]:
```suggestion
-   **Light**: Downloads all block headers, block data, and verifies some randomly. 
```

@holiman My question here is how does someone fetch portions of the state? Or is that best reserved for outside of a getting started guide anyway?

File: docs/_getting-started/index.md (hunk 1 above)
Comment (holiman) [reply docs/_getting-started/index.md]: `state` means 
- All contracts, nonces, balances, storage entries

The state is stored in a merkle trie, and the state at any block can be uniquely identified by the root hash of the tree at that point in time.

So block-by-block generates the state by executing transactions sequentially. Whereas fast-sync downloads the state from peers, and then just validates that the trie root hash indeed corresponds to what the header says that the root-hash should be

File: docs/_getting-started/index.md (hunk 2 above)
Comment (holiman) [reply docs/_getting-started/index.md]: Executing a transaction means
- validate basics, like there's enough balance on the account, 
- do value transfer from sender to recipient, 
- execute code in the ethereum VM, which may mean further internal transfers and storage updates
- deduct gas costs from sender,

File: docs/_getting-started/index.md (hunk 3 above)
Comment (holiman) [reply docs/_getting-started/index.md]: Probably best reserved for outside. But essentially, if you know the root hash, and you are interested in the leaf-values of the trie that correspond to your particular address, you can get a 'stubbed' tree, which contains the leaf you are interested in, and the neighbouring hashes that are needed to validate that the leaf is indeed part of a trie that has this exact root hash

Comment (ChrisChinchilla) [reply docs/_getting-started/index.md]: Righto, noted for next-next-steps, or something…

**Pull request merged**